packages/dicomImageLoader/src/imageLoader/createImage.ts

```diff
@@ -214,27 +223,24 @@ function createImage(
       const sopCommonModule: MetadataSopCommonModule =
         cornerstone.metaData.get('sopCommonModule', imageId) || {};
       if (isColorImage) {
-        if (useRGBA) {
-          // JPEGBaseline (8 bits) is already returning the pixel data in the right format (rgba)
-          // because it's using a canvas to load and decode images.
-          if (
-            !isJPEGBaseline8BitColor(imageFrame, transferSyntax) &&
-            imageFrame.photometricInterpretation !== 'RGB'
-          ) {
-            canvas.height = imageFrame.rows;
-            canvas.width = imageFrame.columns;
-            const context = canvas.getContext('2d');
-            const imageData = context.createImageData(
-              imageFrame.columns,
-              imageFrame.rows
-            );
-
-            convertColorSpace(imageFrame, imageData.data, useRGBA);
-            imageFrame.imageData = imageData;
-            imageFrame.pixelData = imageData.data;
-            imageFrame.pixelDataLength = imageData.data.length;
-          }
-        } else if (isJPEGBaseline8BitColor(imageFrame, transferSyntax)) {
+        const { rows, columns } = imageFrame;
+        if (TRANSFER_SYNTAX_USING_PHOTOMETRIC_COLOR[transferSyntax]) {
+          canvas.height = imageFrame.rows;
+          canvas.width = imageFrame.columns;
+          const context = canvas.getContext('2d');
+          const imageData = context.createImageData(
+            imageFrame.columns,
+            imageFrame.rows
+          );
+
+          convertColorSpace(imageFrame, imageData.data, useRGBA);
+          imageFrame.imageData = imageData;
+          imageFrame.pixelData = imageData.data;
+          imageFrame.pixelDataLength = imageData.data.length;
+        } else if (
+          !useRGBA &&
```

Comment:
i'm surpised that this says `!useRGBA`, but then below in the logic of if statement we have remove A from RGBA

Reply:
I updated the comment below, but this case is when we NEED RGB (that is !useRGBA) but we have RGBA, so we need to convert.